### PR TITLE
Fix broken test for utils.solver with symengine

### PR DESCRIPTION
**Pull request by cdiener**
* [X] fix failing test on newer sympy versions
* [X] description of feature/fix
* [X] tests added/passed
* [X] add an entry to the [next release](../release-notes/next-release.md)

Minor change that repairs a broken test. Also, the logic of the test was a bit off and has now been corrected.

### CODE REVIEW

**Comment by cdiener** [comment]
Only affects symengine, apparently.

File: tests/test_util/test_solver.py [hunk 1]
```diff
@@ -118,14 +118,19 @@ def test_absolute_expression(model: "Model") -> None:
 def test_fix_objective_as_constraint(solver: str, model: "Model") -> None:
     """Test fixing present objective as a constraint."""
     model.solver = solver
+    opt = model.slim_optimize()
     with model as m:
         su.fix_objective_as_constraint(model, 1.0)
         constraint_name = m.constraints[-1]
-        assert abs(m.constraints[-1].expression - m.objective.expression) < 1e-6
+        assert (m.constraints[-1].expression - m.objective.expression).simplify() == 0
+        assert m.constraints[-1].lb == pytest.approx(opt)
```
Comment (Midnighter) [inline tests/test_util/test_solver.py]: It seems better to me, to pass in a name for the constraint, and then use that name to retrieve the constraint object.

Comment (Midnighter) [reply tests/test_util/test_solver.py]: Also, the variable name `constraint_name` is misleading here, as it is in fact the object and not the name.

File: tests/test_util/test_solver.py [hunk 2]
```diff
@@ -118,14 +118,19 @@ def test_absolute_expression(model: "Model") -> None:
...
     assert constraint_name not in m.constraints
     su.fix_objective_as_constraint(model)
     constraint_name = model.constraints[-1]
-    assert abs(model.constraints[-1].expression - model.objective.expression) < 1e-6
+    assert (
+        model.constraints[-1].expression - model.objective.expression
+    ).simplify() == 0
+    assert m.constraints[-1].lb == pytest.approx(opt)
```
Comment (Midnighter) [inline tests/test_util/test_solver.py]: As above.

**Review by Midnighter — CHANGES_REQUESTED** [review]
Thank you for the fix. I would at least change it to

```py
fix_constraint = m.constraints[-1]
```

and then perform assertions on that.

File: tests/test_util/test_solver.py (hunk 1 above)
Comment (cdiener) [reply tests/test_util/test_solver.py]: Yeah, I did not write the original test and did not want to modify more than necessary. I changed it now to be as suggested.

**Review by Midnighter — APPROVED** [review]
Thanks for the changes.

**Pull request merged**